src/test/uint256_tests.cpp

```diff
@@ -386,6 +386,34 @@ BOOST_AUTO_TEST_CASE(from_hex)
     TestFromHex<Wtxid>();
 }
 
+BOOST_AUTO_TEST_CASE(from_user_hex)
+{
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0x").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("00").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("1").value(), uint256::ONE);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0x10").value(), uint256{0x10});
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("10").value(), uint256{0x10});
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0xff").value(), uint256{0xff});
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("ff").value(), uint256{0xff});
+    // a value is expected when the last (65th) character is trimmed off valid_hex_65
+    constexpr std::string_view valid_hex_65{"0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0"}; // 0x prefix and 65 hex chars
+    static_assert(valid_hex_65.size() == 2 + 65);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex(valid_hex_65.substr(2, 64)).value().ToString(), valid_hex_65.substr(2, 64));
+    BOOST_CHECK_EQUAL(uint256::FromUserHex(valid_hex_65.substr(0, 66)).value().ToString(), valid_hex_65.substr(2, 64));
+
+    BOOST_CHECK(!uint256::FromUserHex("0x0 "));                 // no spaces at end,
```

Comment:
Done, and reorganized the test a bit to use `std::string valid_hex_64` instead of `std::string_view valid_hex_65` which makes cat operations significantly less verbose, and makes the test more readable imo.